src/lib/IPFS/ipfs.js

```diff
@@ -0,0 +1,116 @@
+/*!
+ * Copyright(c) ConsenSys Software Inc.
+ * Copyright(c) https://consensys.net/
+ * MIT Licensed
+ */
+/* eslint-disable */
+import { ERROR_LOG, errorLogger } from '../error/handler.js';
+import fs from 'fs';
+import { create as ipfsClient, globSource } from 'ipfs-http-client';
+
+export default class IPFS {
+  #ipfsClient;
+
+  constructor({ ipfsUrl, projectId, projectSecret }) {
+    if (!ipfsUrl) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.Ipfs_constructor,
+          message: ERROR_LOG.message.no_infura_ipfsUrl_supplied,
+        }),
+      );
+    }
+
+    if (!projectId) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.Ipfs_constructor,
+          message: ERROR_LOG.message.no_infura_projectID_supplied,
+        }),
+      );
+    }
+
+    if (!projectSecret) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.Ipfs_constructor,
+          message: ERROR_LOG.message.no_infura_projectSecret_supplied,
+        }),
+      );
+    }
+
+    const auth = 'Basic ' + Buffer.from(projectId + ':' + projectSecret).toString('base64');
+
+    this.#ipfsClient = ipfsClient({
+      url: ipfsUrl,
+      headers: {
+        authorization: auth,
+      },
+    });
+  }
+
+  async uploadFile({ source }) {
```

Comment:
I don't think we need intermediary methods here. You can maybe directly call "save", "saveDirectory" from the exposed method, no?